llvm/lib/TableGen/TGParser.cpp

```diff
@@ -1455,6 +1455,49 @@ const Init *TGParser::ParseOperation(Record *CurRec, const RecTy *ItemType) {
     return (ExistsOpInit::get(Type, Expr))->Fold(CurRec);
   }
 
+  case tgtok::XMatch: {
```

Comment:
Can you use the parsing code below that handles XConcat and all the other binops?